contributing/building-a-bento-amp-extension.md

```diff
@@ -0,0 +1,681 @@
+[amp.dev](https://amp.dev/) on
+[GitHub](https://github.com/ampproject/amp.dev).
+
+## Updating build configs
+
+In order for your element to build correctly you would need to make few
+changes to [`build-system/compile/bundles.config.js`](../build-system/compile/bundles.config.js) to tell it about your
+extension, its files and its examples. You will need to add an entry in the `extensionBundles` array.
+
+```javascript
+exports.extensionBundles = [
+...
+  {name: 'amp-kaltura-player', version: '0.1', latestVersion: '0.1'},
+  {name: 'amp-carousel', version: '0.1', latestVersion: '0.1', options: {hasCss: true}},
+...
+];
+```
+
+Note, if you are providing a version upgrade (pre-existing 0.1 to Bento 1.0, for example), it is important to note that the latest version is the 0.1 until the 1.0 version is no longer experimental and fully launched. The following is the most straight-forward way to specify this:
+
+```javascript
+exports.extensionBundles = [
+...
+  {name: 'amp-my-element', version: ['0.1', '1.0'], latestVersion: '0.1', options: {hasCss: true}},
+...
+];
+```
+
+However, note that if one version is `hasCss: true` and the other is `hasCss: false`, they will require two separate entries:
+
+```javascript
+exports.extensionBundles = [
+...
+  {name: 'amp-my-element', version: '0.1', latestVersion: '0.1', options: {hasCss: true}},
+  {name: 'amp-my-element', version: '1.0', latestVersion: '0.1'},
+...
+];
+```
+
+## Versioning
+
+AMP runtime is currently in v0 major version. Extensions versions are
+maintained separately. If your changes to your non-experimental
+extension makes breaking changes that are not backward compatible you
+should version your extension. This would usually be by creating a 0.2
+directory next to your 0.1.
+
+When version 0.2 is under development, make sure that `latestVersion` is
+set to 0.1 for both the 0.1 and 0.2 entries in `extensionBundles`. Once 0.2
+is ready to be released, `latestVersion` can be changed to 0.2.
+
+If your extension is still in experiments breaking changes usually are
+fine so you can just update the same version.
+
+Note that Bento upgrades to existing AMP components should go by one major version, meaning it should create a 1.0 directory next to an existing 0.1 or 0.2.
+
+## Unit tests
+
+Make sure you write good coverage for your code. We require unit tests
+for all checked in code. We use the following frameworks for testing:
+
+-   [Mocha](https://mochajs.org/), our test framework
+-   [Karma](https://karma-runner.github.io/), our tests runner
+-   [Sinon](http://sinonjs.org/), spies, stubs and mocks.
+
+For faster testing during development, consider using --files argument
+to only run your extensions' tests.
+
+```shell
+$ gulp unit --files=extensions/amp-my-element/0.1/test/test-amp-my-element.js --watch
+```
+
+## Linting and formatting
+
+The following command should be run to validates JS files against the ESLint linter.
+
+```shell
+$ gulp lint --local_changes
```

Comment:
Check with wg-infra (probably Raghu or Esther, but not tagging them so they aren't subscribed to the whole PR), but I think `gulp lint` includes Prettier, so the below command may be redundant

Reply:
Checked with Esther who confirmed, removing the prettier command here. :)